core/controllers/feedback_test.py

```diff
@@ -554,3 +567,206 @@ def test_feedback_threads_with_suggestions(self):
+
+    def setUp(self):
+        super(RecentFeedbackMessagesHandlerTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.set_admins([self.OWNER_USERNAME])
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.exp_id = 'exp_id'
+
+    def test_get_recently_posted_feedback_messages(self):
+        self.login(self.OWNER_EMAIL)
+
+        response = self.get_json(
+            feconf.RECENT_FEEDBACK_MESSAGES_DATA_URL)
+
+        self.assertEqual(response['results'], [])
+        self.assertFalse(response['more'])
+
+        self.save_new_valid_exploration(
+            self.exp_id, self.owner_id, title='Exploration title',
+            category='Architecture', language_code='en')
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_EXPLORATION, self.exp_id, self.owner_id,
+            'a subject', 'some text')
+
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_EXPLORATION, self.exp_id, self.owner_id,
+            'new subject', 'new text')
+
+        response = self.get_json(
+            feconf.RECENT_FEEDBACK_MESSAGES_DATA_URL)
+        results = response['results']
+
+        self.assertEqual(len(results), 2)
+
+        self.assertFalse(response['more'])
+        self.assertEqual(results[0]['text'], 'new text')
+        self.assertEqual(results[0]['updated_subject'], 'new subject')
+        self.assertEqual(results[0]['entity_type'], 'exploration')
+        self.assertEqual(results[0]['entity_id'], self.exp_id)
+
+        self.assertEqual(results[1]['text'], 'some text')
+        self.assertEqual(results[1]['updated_subject'], 'a subject')
+        self.assertEqual(results[1]['entity_type'], 'exploration')
+        self.assertEqual(results[1]['entity_id'], self.exp_id)
+
+        self.logout()
+
+
+class FeedbackStatsHandlerTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(FeedbackStatsHandlerTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.set_admins([self.OWNER_USERNAME])
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.exp_id = 'exp_id'
+
+    def test_get_num_threads_after_creating_feedback_analytics(self):
+        self.login(self.OWNER_EMAIL)
+
+        self.get_json(
+            '%s/%s' % (feconf.FEEDBACK_STATS_URL_PREFIX, self.exp_id),
+            expected_status_int=404)
+
+        self.save_new_valid_exploration(
+            self.exp_id, self.owner_id, title='Exploration title',
+            category='Architecture', language_code='en')
+
+        response = self.get_json(
+            '%s/%s' % (feconf.FEEDBACK_STATS_URL_PREFIX, self.exp_id))
+
+        self.assertEqual(response['num_total_threads'], 0)
+        self.assertEqual(response['num_open_threads'], 0)
+
+        feedback_services.create_thread(
+            'exploration', self.exp_id, self.owner_id, 'subject', 'text')
+
+        with self.swap(
```

Comment:
Please clean up this block, it's very messy:

```python
with self.swap(feedback_jobs_continuous
               'FeedbackAnalyticsAggregator',
               MockFeedbackAnalyticsAggregator):
  (feedback_jobs_continuous.FeedbackAnalyticsAggregator
      .start_computation())

Reply:
Done